test/extended/util/framework.go

```diff
@@ -2106,3 +2116,47 @@ func DoesApiResourceExist(oc *CLI, apiResourceName string) (bool, error) {
 
 	return false, nil
 }
+
+var (
+	microShiftVersion string
+	microShiftMutex   sync.Mutex
+)
+
+func getMicroShiftVersion(ocClient *CLI) (string, error) {
+	microShiftMutex.Lock()
+	defer microShiftMutex.Unlock()
+
+	if microShiftVersion == "" {
+		cm, err := ocClient.AdminKubeClient().CoreV1().ConfigMaps("kube-public").Get(context.Background(), "microshift-version", metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return "", nil
+			}
+			return "", fmt.Errorf("unable to get microshift version configmap: %w", err)
+		}
+		if major, ok := cm.Data["major"]; !ok {
+			return "", fmt.Errorf("major not found")
+		} else {
+			microShiftVersion = fmt.Sprintf("%s", major)
+		}
+		if minor, ok := cm.Data["minor"]; !ok {
+			return "", fmt.Errorf("minor not found")
+		} else {
+			microShiftVersion = fmt.Sprintf("%s.%s", microShiftVersion, minor)
+		}
+		if version, ok := cm.Data["version"]; !ok {
```

Comment:
The version string is going to be the full display version, so you could just return that instead of concatenating it with the major and minor versions.

Reply:
Mind if we change it in a future commit? As of today the version comes from the git version. It is true though that the `microShiftVersion` is only there for caching and (maybe) future convenience to extract the concrete version to do something with it.
In this change the caller is not able to see the version, only if its microshift or not.

Reply:
Fixed it, saw it was including the full version when compiling.